pkg/metrics/prometheus.go

```diff
@@ -0,0 +1,152 @@
+package metrics
+
+import (
+	"github.com/grafana/grafana/pkg/log"
+	"github.com/prometheus/client_golang/prometheus"
+	"net/http"
+	"strings"
+)
+
+const (
+	namespace = "grafana"
+	subsystem = "backend"
+)
+
+type PrometheusPublisher struct {
+	CounterCollectors map[string]*prometheus.CounterVec
+	GaugeCollectors   map[string]*prometheus.GaugeVec
+}
+
+func CreatePrometheusPublisher() (*PrometheusPublisher, error) {
+	go func() {
+		http.Handle("/metrics", prometheus.Handler())
+		err := http.ListenAndServe(":9200", nil)
```

Comment:
> We're talking averages here over long time periods so peaks don't matter, unless they happen to be right at the start/end of the period.

No, that's exactly what I am not talking about. I wan accurate sums over long time periods (like total logins per hour/day/week). There should not be a need to do any average in that calculation. 

Did not know about the `increase` function. Would be interested to know how it works (how it does the sum over time of the raw counter given how the step parameter works I am not sure how this can be very accurate but I am saying this from complete ignorance how the internals of Promethus works). 

> Let's say you've got a delta every minute, and they happen to be coming in at 20 seconds past the minute. The data point in each minute slice does not tell you the amount in that minute, rather it's (on average) 66% of the previous minute and 33% this minute. Thus if you were to do say a sum of an hours data, then you'd be missing 40 seconds of events and also including 40 seconds of events that you shouldn't be. Thus it is inaccurate.

That's an inaccuracy that is very seldom important, I mean you do not loose counts they will just be attributed to the another minute. And when looking at larger aggregations they become even less important. 

A more realistic example would be sending data to Graphite every second or every 10 seconds, which aggregates that (using sum) to 1min buckets for data more than 1 day old, to 10min buckets for data more than 7 days old. Not seeing any significant inaccuracies here. And then when doing queries that aggregate data let's say logins per day, summarize(logins.count, '1d'), not seeing how that will be inaccurate.